app/src/main/java/at/gammastrahlung/monopoly_app/activities/BoardActivity.java

```diff
@@ -485,11 +496,72 @@ public void moveAvatarAfterCheating() {
     @Override
     public void onForward(int value) {
         moveAvatar();
+
+        Player currentPlayer = GameData.getGameData().getCurrentPlayer();
+       int currentFieldIndex = currentPlayer.getCurrentFieldIndex();
+        int moveValue = value+currentFieldIndex;
+        followingIndex = (moveValue) % 40;
+        MonopolyClient.getMonopolyClient().sendCurrentFieldInfo(followingIndex);
     }
 
     // Player does want to cheat and moves a selected value forward
     @Override
     public void onSelectedValue(int value){
         moveAvatarAfterCheating();
+        Player currentPlayer = GameData.getGameData().getCurrentPlayer();
+        int currentFieldIndex = currentPlayer.getCurrentFieldIndex();
+        int moveValue = value+currentFieldIndex;
+        followingIndex = (moveValue) % 40;
+        MonopolyClient.getMonopolyClient().sendCurrentFieldInfo(followingIndex);
+    }
+
+
+
+
+        public void activatePurchaseDialog(View view) {
+
+            PurchaseDialogFragment purchaseDialogFragment = PurchaseDialogFragment.newInstance();
+
+
+            purchaseDialogFragment.setPurchaseDialogListener(new PurchaseDialogFragment.PurchaseDialogListener() {
+                @Override
+                public void onNoButtonClicked() {
+
+                    }
+
+                @Override
+                public void onYesButtonClicked() {
+
+                                    }
+            });
+
+
+            purchaseDialogFragment.show(getSupportFragmentManager(), "PurchaseDialogFragment");
+        }
+
+    public void activateAuctionDialog(View view) {
+        AuctionDialogFragment auctionDialogFragment = AuctionDialogFragment.newInstance();
+        auctionDialogFragment.show(getSupportFragmentManager(), "auctionDialog");
```

Comment:
This should be executed on UI thread.
The only call to this method is in WebSocketHandler which runs on the thread created by OkHttp